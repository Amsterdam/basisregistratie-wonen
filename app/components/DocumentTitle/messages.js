import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  default_title: {
    id: `${scope}.registration_plural`,
  },
});
