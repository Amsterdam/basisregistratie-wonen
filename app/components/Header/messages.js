/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
  },
  log_in: {
    id: `${scope}.log_in`,
  },
  log_in_adw: {
    id: `${scope}.log_in_adw`,
  },
  log_out: {
    id: `${scope}.log_out`,
  },
});
