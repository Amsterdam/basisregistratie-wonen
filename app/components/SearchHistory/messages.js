/*
 * SearchHistory Messages
 *
 * This contains all the text for the SearchHistory component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_history_header: {
    id: `${scope}.search_history_header`,
    defaultMessage: 'Recently search for',
  },
});
