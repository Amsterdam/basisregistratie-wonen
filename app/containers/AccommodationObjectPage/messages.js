/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
