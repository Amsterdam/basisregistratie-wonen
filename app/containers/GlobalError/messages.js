/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
