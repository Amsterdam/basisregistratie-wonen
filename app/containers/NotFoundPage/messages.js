/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  document_title: {
    id: `${scope}.registration_plural`,
  },
  header: {
    id: `${scope}.page_not_found`,
  },
});
