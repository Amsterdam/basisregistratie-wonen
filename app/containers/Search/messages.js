import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_placeholder: {
    id: `${scope}.search_placeholder`,
  },
});
