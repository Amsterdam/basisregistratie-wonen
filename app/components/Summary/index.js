import React, { Fragment } from 'react';
import { intlShape } from 'react-intl';
import styled from 'styled-components';

import { summaryPropType } from 'utils/propTypes';
import SectionHeading from 'components/SectionHeading';
import messages from 'containers/App/messages';
import LoadingIndicator from 'components/LoadingIndicator';

const Dt = styled.dt`
  font-family: Avenir Next LT W01 Demi;
`;

const Dd = styled.dd`
  padding-left: 1em;
`;

const Summary = ({ data, intl: { locale, formatMessage } }) => {
  if (!data) return null;

  const numItems = Object.keys(data).length;

  if (!numItems) return <LoadingIndicator />;

  return (
    <>
      <SectionHeading>{formatMessage(messages.overview)}</SectionHeading>
      <dl>
        {Object.keys(data).map(key => (
          <Fragment key={key}>
            <Dt lang={locale}>{formatMessage(data[key].label)}</Dt>
            <Dd>{data[key].value}</Dd>
          </Fragment>
        ))}
      </dl>
    </>
  );
};

Summary.propTypes = {
  data: summaryPropType,
  intl: intlShape.isRequired,
};

Summary.defaultProps = {
  data: null,
};

export default Summary;
