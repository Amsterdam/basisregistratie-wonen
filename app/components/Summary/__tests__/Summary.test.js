import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';

import messages from '../../../translations/nl.json';
import Summary from '..';

const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

const summaryData = {
  address: {
    label: {
      id: 'registraties.address',
    },
    value: ['FakeStreetName 229-X', ' ', '1000XX'],
  },
  public_space_id: {
    label: {
      id: 'registraties.public_space_id',
    },
    value: '0363300000004297',
  },
  number_identification_id: {
    label: {
      id: 'registraties.number_identification.id',
    },
    value: '0363200000201567',
  },
  accommodation_object_id: {
    label: {
      id: 'registraties.accommodation_object.id',
    },
    value: '0363010000740965',
  },
  house_id: {
    label: {
      id: 'registraties.house.id',
    },
    value: '0363100012070761',
  },
  cadastral_object_nr: {
    label: {
      id: 'registraties.cadastral_object.indication',
    },
    value: 'ASD99 K 01337 A 1337',
  },
  chamber_of_commerce_nr: {
    label: {
      id: 'registraties.chamber_of_commerce.nr',
    },
    value: '33006516',
  },
  RSIN: {
    label: {
      id: 'registraties.rsin',
    },
    value: '2562856',
  },
};

describe('Summary', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const numSummaryItems = Object.keys(summaryData).length;
    const { container } = render(<Summary intl={intl} data={summaryData} />);

    expect(container.firstChild).toMatchSnapshot();
    expect(document.getElementsByTagName('dl')).toHaveLength(1);
    expect(document.getElementsByTagName('dd')).toHaveLength(numSummaryItems);
    expect(document.getElementsByTagName('dt')).toHaveLength(numSummaryItems);
  });

  it('should render null', () => {
    const { container, rerender, getByTestId } = render(<Summary intl={intl} />);

    expect(container.firstChild).toBeNull();

    rerender(<Summary intl={intl} data={{}} />);

    expect(getByTestId('progress-wrapper')).not.toBeNull();
  });
});
