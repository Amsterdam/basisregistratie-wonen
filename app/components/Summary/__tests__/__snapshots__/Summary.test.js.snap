// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary matches the snapshot 1`] = `
.c0 {
  margin-top: 30px;
}

@media print {
  .c0 {
    margin-top: 0;
  }
}

<h3
  class="c0"
>
  Overzicht
</h3>
`;
